Release 0.8.0

blakeblackshear opened the pull request.

blakeblackshear committed:
update events model

blakeblackshear committed:
store events in tinydb

blakeblackshear committed:
basic database model and api endpoint

blakeblackshear committed:
saving events and simple endpoint

blakeblackshear committed:
move primary script into the module

blakeblackshear committed:
app container and config schema

blakeblackshear committed:
init db/http/mqtt

blakeblackshear committed:
add detector processes

blakeblackshear committed:
add detected_frames_processor

blakeblackshear committed:
add camera processors

blakeblackshear committed:
add capture processes

blakeblackshear committed:
add event processor

blakeblackshear committed:
add back all endpoints

blakeblackshear committed:
add watchdog

blakeblackshear committed:
config setup

blakeblackshear committed:
add nginx and change default file locations

blakeblackshear committed:
create typed config classes

blakeblackshear committed:
use a queue for logging

blakeblackshear committed:
naming threads and processes for logs

blakeblackshear committed:
sort imports

blakeblackshear committed:
maintain thumbnail frames for tracked objects

blakeblackshear committed:
create tracked object class and save thumbnails

blakeblackshear committed:
stop writing json file to disk

blakeblackshear committed:
update nginx config

blakeblackshear committed:
fixes

blakeblackshear committed:
reduce logging

blakeblackshear committed:
cleanup

blakeblackshear committed:
don't save thumbnails for false positives

blakeblackshear committed:
ensure some valid thumbnail is available

blakeblackshear committed:
cleanup false_positive attribute

blakeblackshear committed:
better frame handling for best images

blakeblackshear committed:
attempt to fix missing thumbs

blakeblackshear committed:
dont use a property

blakeblackshear committed:
add some debug logging to frame cache

blakeblackshear committed:
only set thumbnail data if object is a true positive

blakeblackshear committed:
check for None value thumbnail_data

blakeblackshear committed:
add bas64 encoded thumbnail to the database

blakeblackshear committed:
only save events when a clip is created

blakeblackshear committed:
add parameters to event query

blakeblackshear committed:
new http endpoints

blakeblackshear committed:
use localtime in group by

blakeblackshear committed:
add clip retention to config

blakeblackshear committed:
add event cleanup thread

blakeblackshear committed:
manage events for unlisted cameras

blakeblackshear committed:
sleep in the right place

blakeblackshear committed:
add zones to summary data

blakeblackshear committed:
add service to get by id

blakeblackshear committed:
add endpoint for event thumbnail

blakeblackshear committed:
pull from memory if event in progress

blakeblackshear committed:
publish events like a change feed

blakeblackshear committed:
ensure all events are cleaned up

blakeblackshear committed:
only publish end events for true positives

blakeblackshear committed:
oops

blakeblackshear committed:
minimize logging

blakeblackshear committed:
avoid null error

rb2k committed:
Add support for AMD Ryzen iGPU (fixes #311)

This package will add support for the iGPU of AMD Ryzen and presumably a few more AMD cards.
See details of the package here: https://packages.ubuntu.com/focal/mesa-va-drivers
It also adds support for the open source Nvidia Nouveau driver according to https://wiki.debian.org/HardwareVideoAcceleration

no2chem committed:
Use cv2.bitwise_and instead of numpy.where

blakeblackshear committed:
add support for rebroadcasting as rtmp

blakeblackshear committed:
fix fontconfig error

blakeblackshear committed:
add multiple streams per camera

blakeblackshear committed:
graceful exit of subprocesses

blakeblackshear committed:
ensure zones dont have the same name as a camera

blakeblackshear committed:
fix log timeout

blakeblackshear committed:
add record settings to config

blakeblackshear committed:
add recording maintenance

blakeblackshear committed:
serve up recordings with nginx

blakeblackshear committed:
cleanup empty directories

blakeblackshear committed:
make directories constants

blakeblackshear committed:
publish object counts rather than on/off

blakeblackshear committed:
reduce min timestamp size

blakeblackshear committed:
optional android notification aspect ratio

blakeblackshear committed:
add zeroconf discovery

blakeblackshear committed:
exit on config errors

blakeblackshear committed:
add logging for directory creation

blakeblackshear committed:
default save_clips objects

blakeblackshear committed:
dont delete the recordings directory

blakeblackshear committed:
no need to write jpg disk

blakeblackshear committed:
log level configuration

blakeblackshear committed:
update some default config values

blakeblackshear committed:
doc updates

blakeblackshear committed:
fix fontconfig issue

blakeblackshear committed:
fix default args

blakeblackshear committed:
better exception handling

blakeblackshear committed:
fix graceful exits

blakeblackshear committed:
fix flask logger config

blakeblackshear committed:
comment you zeroconf

blakeblackshear committed:
set ffmpeg image versions

blakeblackshear committed:
readme updates

blakeblackshear committed:
tweak screenshots

blakeblackshear committed:
readme updates

blakeblackshear committed:
clarify height width and fps

blakeblackshear committed:
catch all psutil errors

blakeblackshear committed:
camera recommendations

blakeblackshear committed:
handle and warn if roles dont match enabled features

blakeblackshear committed:
readme cleanup fixes #332

pdecat committed:
Document beta addon host

tubalainen committed:
Updated docker command line...

...to correspond with 0.8.0 feature set.

blakeblackshear committed:
make shm consistent with compose

blakeblackshear committed:
add model dimensions to config

blakeblackshear committed:
allow defining model shape and switch to mobiledet as default model

blakeblackshear committed:
updating compose example to reduce confusion

blakeblackshear committed:
swap width and height to reduce confusion

blakeblackshear committed:
allow region to extend beyond the frame

blakeblackshear committed:
handle empty string args

blakeblackshear committed:
fix process clip

tubalainen committed:
Updated file

ref: https://github.com/blakeblackshear/frigate/issues/373

blakeblackshear committed:
add telegram example

blakeblackshear committed:
allow db path to be customized

blakeblackshear committed:
allow process clips to output a csv of scores

blakeblackshear committed:
switch default threshold to .7

blakeblackshear committed:
fix test

blakeblackshear committed:
update gitignore

pdecat commented on docker/Dockerfile.base:
```diff
@@ -27,18 +28,25 @@ RUN apt-get -qq update \
-RUN wget -q https://github.com/google-coral/edgetpu/raw/$MODEL_REFS/test_data/ssd_mobilenet_v2_coco_quant_postprocess.tflite -O /cpu_model.tflite
-
-RUN mkdir /cache /clips
+RUN wget -q https://github.com/google-coral/test_data/raw/master/ssdlite_mobiledet_coco_qat_postprocess_edgetpu.tflite -O /edgetpu_model.tflite
```
Shouldn't the labelmap.txt file be updated from https://github.com/google-coral/test_data/raw/master/coco_labels.txt too?

There's this notable diff:

```
diff <(curl -s https://raw.githubusercontent.com/blakeblackshear/frigate/master/labelmap.txt) \
>    <(curl -s https://raw.githubusercontent.com/google-coral/test_data/master/coco_labels.txt)
8c8
< 7  car
---
> 7  truck
```

blakeblackshear replied:
That is intentional. The model will often flip flop between cars and trucks. By naming them the same, it prevents each label change from triggering a new object id to track.

blakeblackshear committed:
configurable motion and detect settings

blakeblackshear committed:
adding version endpoint

blakeblackshear committed:
allow the mask to be a list of masks

blakeblackshear committed:
allow runtime drawing settings for mjpeg and latest

blakeblackshear committed:
re-crop to the object rather than the region

blakeblackshear committed:
add post_capture option

blakeblackshear committed:
optimize clips fixes #299

blakeblackshear committed:
add num_threads fixes #322

blakeblackshear committed:
add qsv support to amd64 image

blakeblackshear committed:
handle scenario with empty cache

blakeblackshear committed:
readme updates

blakeblackshear committed:
publish event updates on zone change

blakeblackshear committed:
update process clip for latest changes

blakeblackshear committed:
docs and issue template

blakeblackshear committed:
enable mounting tmpfs volume on start

blakeblackshear committed:
use CACHE_DIR constant

blakeblackshear committed:
remove redundant error output

blakeblackshear committed:
fix tmpfs

blakeblackshear committed:
better mask error handling

blakeblackshear committed:
add clips endpoint to readme

kluszczyn committed:
Recordings - fix expire_file

blakeblackshear committed:
add jpg snapshots to disk and clean up config

blakeblackshear committed:
cleanup save_Clips/clips inconsistency

blakeblackshear committed:
enable turning clips on and off via mqtt

blakeblackshear committed:
turn off snapshots via mqtt

blakeblackshear committed:
add database migrations

blakeblackshear committed:
store has_clip and has_snapshot on events

blakeblackshear committed:
handle process exit exceptions

ljmerza committed:
add notes for Blue Iris RTSP support

yllar committed:
Update README.md

change tmpfs size from 100MB to 1GB

Commit:
Set titles for forked processes

Commit:
New stats module, refactor stats generation out of http module.

StatsEmitter thread to send stats to MQTT every 60 seconds by default, optional stats_interval config value.

New service stats attribute, containing uptime in seconds and version.

paularmstrong committed:
feat!: web user interface

paularmstrong committed:
fix: ensure timestamp is drawn above mask

blakeblackshear committed:
init variables on camera state

blakeblackshear committed:
add retention settings for snapshots

blakeblackshear committed:
try to cleanup some migration logging

paularmstrong committed:
fix(web): ensure postcss and postcss-cli are marked as deps

pdecat committed:
Add missing migrations in docker images

pdecat committed:
Fix Makefile to ignore gpg signatures in commits

blakeblackshear committed:
add missing global shapshots config

blakeblackshear committed:
add object masks and move moton mask

blakeblackshear committed:
add mask as object filter

blakeblackshear committed:
handle null thumbnail data

blakeblackshear committed:
switch to docker based web builds

blakeblackshear committed:
move setproctitle to prebuilt wheel location

paularmstrong committed:
fix(web): mask zone editor to handle object filter masks

Includes additional handlers for adding/removing masks, as well as click to copy configs

fixes #523

blakeblackshear committed:
enable and disable detection via mqtt

blakeblackshear committed:
add env vars to config

fixes #509

blakeblackshear committed:
add detection enable to config

fixes #482

blakeblackshear committed:
ensure each camera has a detect role set

blakeblackshear committed:
add change type to events topic

#476

blakeblackshear committed:
don't fallback to the CPU

fixes #381

blakeblackshear committed:
remove support for image masks

blakeblackshear committed:
update object filters to inherit like motion settings

blakeblackshear committed:
snapshots config typo

blakeblackshear committed:
update readme

blakeblackshear committed:
allow summary data to be filtered

blakeblackshear committed:
mqtt tweaks for switches

paularmstrong committed:
fix(web): ensure mask editing works in firefox

blakeblackshear committed:
rename snapshot endpoint to thumbnail

blakeblackshear committed:
docs updates for notification changes

paularmstrong committed:
feat(web): hash build files to avoid cache issues

blakeblackshear committed:
move wheels to build container

blakeblackshear committed:
versioning wheels image

blakeblackshear committed:
no longer need special aarch64 wheels build

blakeblackshear committed:
fix zone config

paularmstrong committed:
fix(web): ensure button bg colors show in prod builds

paularmstrong committed:
feat(web): layout & auto-update debug page

blakeblackshear committed:
pin numpy

blakeblackshear committed:
update wheels version

blakeblackshear committed:
initialize detection correctly from config

blakeblackshear committed:
fix mqtt switch handling

blakeblackshear committed:
update notification example

paularmstrong committed:
fix(web): dark mode text color fixes

fixes #544

blakeblackshear committed:
ensure error message with missing config is printed

blakeblackshear committed:
first pass at subfilter for ingress support

blakeblackshear committed:
update index.js to use baseUrl

blakeblackshear committed:
handle exception when frame isnt in cache

paularmstrong committed:
fix(web): set default path to cameras view

blakeblackshear committed:
fix initial switch state

paularmstrong committed:
feat(web): AutoUpdatingCameraImage to replace MJPEG feed

paularmstrong committed:
docs: move docs to docusaurus

blakeblackshear committed:
dont wait forever for the cache

blakeblackshear committed:
get the thumbnail instead of the full frame

blakeblackshear committed:
add snapshot endpoint that works during the event fixes #575

blakeblackshear committed:
rate limit tracked object updates to every 5 seconds

blakeblackshear committed:
readme update

blakeblackshear committed:
fix table in docs

blakeblackshear committed:
if detection stopped, assume the container needs a restart

blakeblackshear committed:
add debug log when cache is cleaned up

jamescarlos committed:
Update documentation link in sidebar to new docs

blakeblackshear committed:
use sqlitequeuedb

blakeblackshear committed:
tweaking the docs

paularmstrong committed:
fix(web): make camera latest.jpg responsive

paularmstrong committed:
fix(web): ensure all links on events page include pathname

blakeblackshear committed:
add search to docs

blakeblackshear committed:
updating for docusaurus2 docs

paularmstrong committed:
fix(web): object mask editing not showing points

paularmstrong committed:
feat(web): make it possible to add to object masks

paularmstrong committed:
refactor(web): responsive images on content size, throttle AutoUpdatingCameraImage
